remo/dashboard/tests/test_models.py

```diff
@@ -460,3 +463,41 @@ def test_update_event_owner(self):
                                           object_id=event.id)
         eq_(items.count(), 1)
         eq_(items[0].user, new_owner)
+
+
+class ReportActionItems(RemoTestCase):
+    def test_verify_activity(self):
+        model = ContentType.objects.get_for_model(NGReport)
+        items = ActionItem.objects.filter(content_type=model)
+        ok_(not items.exists())
+
+        activity = ActivityFactory.create(name=RECRUIT_MOZILLIAN)
+        mentor = UserFactory.create()
+        user = UserFactory.create(groups=['Rep'], userprofile__mentor=mentor)
+        report = NGReportFactory.create(activity=activity, user=user,
+                                        mentor=mentor)
+
+        items = ActionItem.objects.filter(content_type=model,
+                                          object_id=report.id)
+        eq_(items.count(), 1)
+
+    def test_resolve_verify_action_item(self):
+        model = ContentType.objects.get_for_model(NGReport)
+        items = ActionItem.objects.filter(content_type=model)
+        ok_(not items.exists())
+
+        activity = ActivityFactory.create(name=RECRUIT_MOZILLIAN)
+        mentor = UserFactory.create()
+        user = UserFactory.create(groups=['Rep'], userprofile__mentor=mentor)
+        report = NGReportFactory.create(activity=activity, user=user,
+                                        mentor=mentor)
+
+        items = ActionItem.objects.filter(content_type=model,
+                                          object_id=report.id)
+        eq_(items.count(), 1)
+        report.verified_activity = True
+        report.save()
+
+        for item in items:
```

Comment:
I think that you need to query again the DB to make sure you got up-to-date objects.after ``report.save()``.